### DISCUSSION TITLE
Dag callbacks execution not happening for some dag runs

### DISCUSSION THREAD

**Discussion by iw-pavan** [apache/airflow#35113] [General] [answered]
related to https://github.com/apache/airflow/issues/19767

We have some 15k dag files in dag dir so we have increased "min_file_process_interval" to 1 day. with dag list interval to 10 sec.
In code I see dag callbacks depends on parsing and process which is polling file_path_queue.

Lets assume I've parsed dag file and it wont be parsed for next 24 hours will callbacks wont execute for next 24hours ?

**Comment by potiuk** [comment]
Setting the interval to 1 day is a bad idea. And callbacks is one of the reasons. There are cases where they won't be called - when callbacks cannot be executed in the worker processes - either because they are not task bound or when tasks were killed externally. That's why min_parsing_interval should not be that long.

**Reply by iw-pavan** [reply]
Ohk, I was following best practices mentioned in doc 
https://airflow.apache.org/docs/apache-airflow/stable/faq.html#when-there-are-a-lot-1000-of-dag-files-how-to-speed-up-parsing-of-new-files

Where its mentioned we can set higher than 100minutes. But if callbacks are not going to execute during that configured time then its issue. 
Is it possible to add file to queue if there are any pending callbacks

@potiuk

**Reply by potiuk** [reply, marked as answer]
No - not currently Dag file processor uses: https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#file-parsing-sort-mode and it will not re-parse files that have outstanding callbacks "out of sequence" or "faster". 

But your idea is not bad, I think, in general as proposal - If you would like to analyse it in details propose another scheme there and design of callback execution - you probably would have to prepare, discuss, get approval and implement a new Airflow Improvement Proposal for it https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals - like any other proposals.

**Comment by pavansharma36** [comment]
@potiuk 

It's simple fix, does it need to go through feature discussion
https://github.com/pavansharma36/airflow/commit/84aa8457d90f607899d9572c055270fd7800327d

**Reply by potiuk** [reply]
Surely. The "fix" you mentions has likely many side effects that you do not realize. It will likely break a lot of things for a lot of people.  Callback is not the onlny reason for re-parsing. Reparsing of the dag is also done to potentially regenerate the DAG structure and reserialize it to the database periodically. Your proposal might have undesirable side effects.

If you wish to discuss it - simply explain your solution, and assessment of the side-effect to the devlist.
